### ISSUE TITLE
Runtime Error: Legacy model format is not supported on mobile.

### ISSUE THREAD

**Issue by ppxwwtaking** [pytorch/pytorch#97549]
Recently I met the problem as attached when using relion_class_ranker.py on our new EXXACT workstation to automatically pick the good 2D classes on Relion-4.0, a cryo-EM image processing package. I never had this problem before when using AWS node where relion-4.0 was complied by myself. 

The error seems happening when loading the pre-trained model. My torch version is 1.13.0+cu117. 

The model is class_ranker_0.1.3_torch_1.0.1.pt and renamed as relion_class_ranker_default_model.pt when compiling the relion 4.0 package. 

The python script:

import os
import argparse
import sys

if (sys.version_info < (3, 0)):
    raise Exception('This script supports Python 3 or above.')
    print(end="")  # This script requires Python 3. A Syntax error here means you are running it in Python 2.

try:
    import torch
except ImportError:
    print("PYTHON ERROR: The required python module 'torch' was not found.")
    exit(1)

try:
    import numpy as np
except ImportError:
    print("PYTHON ERROR: The required python module 'numpy' was not found.")
    exit(1)

parser = argparse.ArgumentParser()

parser.add_argument('model_path', type=str)
parser.add_argument('project_dir', type=str)
args = parser.parse_args()

project_dir = args.project_dir
model_fn = args.model_path
feature_fn = os.path.join(project_dir, "features.npy")
images_fn = os.path.join(project_dir, "images.npy")

model = torch.jit.load(model_fn)    ###failed here
features = np.load(feature_fn)
images = np.load(images_fn)

count = features.shape[0]

features_tensor = torch.Tensor(features)
images_tensor = torch.unsqueeze(torch.Tensor(images), 1)
score = model(images_tensor, features_tensor).detach().cpu().numpy()

for i in range(count):
    print(score[i, 0], end=" ")

![Capture](https://user-images.githubusercontent.com/105030617/227603010-9eea714d-026c-4f88-962b-d7f44c327e0f.PNG)

**Comment by pat749** [comment]
The error seems to be related to loading the pre-trained model using **torch.jit.load()**. One possible solution could be to add the model path to the **sys.path** list before importing **torch**, as follows:

```
import os
import argparse
import sys

if (sys.version_info < (3, 0)):
    raise Exception('This script supports Python 3 or above.')
print(end="") # This script requires Python 3. A Syntax error here means you are running it in Python 2.

model_path = 'path/to/model'
sys.path.append(model_path)

try:
    import torch
except ImportError:
    print("PYTHON ERROR: The required python module 'torch' was not found.")
    exit(1)

try:
    import numpy as np
except ImportError:
    print("PYTHON ERROR: The required python module 'numpy' was not found.")
    exit(1)

parser = argparse.ArgumentParser()

parser.add_argument('model_path', type=str)
parser.add_argument('project_dir', type=str)
args = parser.parse_args()

project_dir = args.project_dir
model_fn = args.model_path
feature_fn = os.path.join(project_dir, "features.npy")
images_fn = os.path.join(project_dir, "images.npy")

model = torch.jit.load(model_fn)
features = np.load(feature_fn)
images = np.load(images_fn)

count = features.shape[0]

features_tensor = torch.Tensor(features)
images_tensor = torch.unsqueeze(torch.Tensor(images), 1)
score = model(images_tensor, features_tensor).detach().cpu().numpy()

for i in range(count):
    print(score[i, 0], end=" ")
```

Replace **path/to/model** with the actual path to the directory containing the pre-trained model file.

**Comment by ppxwwtaking** [comment]
Thanks. I will try it. Now I just downgrade to torch 1.9.1 and it works.

Weimin

**Comment by dagitses** [comment]
Thanks Durgesh.

Weimin, please reopen if there's anything to do here on the PyTorch side.
